### DISCUSSION TITLE
[Question] What is the difference between `rehydrate` and `hydrate`

### DISCUSSION THREAD

**Discussion by nstfkc** [vercel/next.js#13606] [Help] [answered]
Hello everyone,
> Disclaimer: I am asking this question here because these concepts are heavily mentioned in `ssr` related topics.  

I am trying to wrap my head around to understand how ssr works in general, but whenever I read an article about it I see `hydration` or `rehydration` keywords. What is the difference between them. 

Thanks!

**Comment by jamesmosier** [comment, marked as answer]
I think often they are used interchangeably throughout the React/SSR community.

(re)hydration is when you have a server rendered HTML document and on the client side React attempts to attach event listeners to the existing markup.

[Put another way](https://addyosmani.com/blog/rehydration/):

> Rehydration in isomorphic web apps involves re-generating state client-side used to render a page on the server. After rendering HTML, hydration is processing inline JSON state stores client-side (e.g Redux state) for data already in the page and attaching DOM event listeners for the experience.

**Reply by nstfkc** [reply]
Thanks for the reply 👍
